module3/src/MyList.java

```diff
@@ -0,0 +1,697 @@
+     * @throws NullPointerException if the specified collection contains one
+     *         or more null elements and this list does not permit null
+     *         elements, or if the specified collection is null
+     * @throws IllegalArgumentException if some property of an element of the
+     *         specified collection prevents it from being added to this list
+     * @throws IndexOutOfBoundsException if the index is out of range
+     *         (<tt>index &lt; 0 || index &gt; size()</tt>)
+     */
+    @Override
+    public boolean addAll(int index, Collection<? extends T> c) {
+        throw new UnsupportedOperationException(); //return false; //Stanislav said we do not need to implement it
+    }
+
+    /**
+     * Removes all of the elements from this list (optional operation).
+     * The list will be empty after this call returns.
+     *
+     * @throws UnsupportedOperationException if the <tt>clear</tt> operation
+     *         is not supported by this list
+     */
+    @Override
+    public void clear() {
+        for (int i = 0; i < size; i++) {
+            this.elementData[i] = null;
+        }
+        this.size = 0;
+        this.capacity = DEFAULT_CAPACITY;
+    }
+
+    /**
+     * Returns the element at the specified position in this list.
+     *
+     * @param index index of the element to return
+     * @return the element at the specified position in this list
+     * @throws IndexOutOfBoundsException if the index is out of range
+     *         (<tt>index &lt; 0 || index &gt;= size()</tt>)
+     */
+    @Override
+    public T get(int index) {
+        if (index < size) {
+            return (T) this.elementData[index];
+        } else {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    /**
+     * Replaces the element at the specified position in this list with the
+     * specified element (optional operation).
+     *
+     * @param index index of the element to replace
+     * @param element element to be stored at the specified position
+     * @return the element previously at the specified position
+     * @throws UnsupportedOperationException if the <tt>set</tt> operation
+     *         is not supported by this list
+     * @throws ClassCastException if the class of the specified element
+     *         prevents it from being added to this list
+     * @throws NullPointerException if the specified element is null and
+     *         this list does not permit null elements
+     * @throws IllegalArgumentException if some property of the specified
+     *         element prevents it from being added to this list
+     * @throws IndexOutOfBoundsException if the index is out of range
+     *         (<tt>index &lt; 0 || index &gt;= size()</tt>)
+     */
+    @Override
+    public T set(int index, T element) {
+        if (index < size) {
+            T previous = (T) this.elementData[index];
+            this.elementData[index] = element;
```

Comment:
what if I have list with elements [1, 2, 3, 4, 5]
and I would like to set one more value to index 3 and value 4.
Usually result will be [1, 2, 3, 4, 4, 5]. Using ArrayList,
But your list will have  [1, 2, 3, 4, 5] as a result.
Please fix it

Reply:
Wow!
I followed the requirement for the "set" method:
     * Replaces the element at the specified position in this list with the
     * specified element (optional operation).
     ...
     * @return the element previously at the specified position
     
I have tested how ArrayList works. – It works same way:
```
        List<Integer> arrayList = new ArrayList<>();
        arrayList.add(1);
        arrayList.add(2);
        arrayList.add(3);
        arrayList.add(4);
        arrayList.add(5);
        System.out.println("arrayList = " + arrayList);

        arrayList.set(3,4);
        System.out.println("arrayList after arrayList.set(3,4) = " + arrayList);
```
Output:
arrayList = [1, 2, 3, 4, 5]
arrayList after arrayList.set(3,4) = [1, 2, 3, 4, 5]

Reply:
added code to compare `set` method behavior for ArrayList and MyList